compiler/test/runnable/placenew.d

```diff
@@ -106,6 +106,32 @@ void test7()
     assert(t.x == 10 && t.y == 20);
 }
 
+/*************************************************/
+// https://github.com/dlang/dmd/issues/21203
+
+struct Y8
+{
+    int a;
+    this() @disable; // default disabled
+    this(int i) { a = i; }
+}
+
+struct S8
+{
+    Y8 m;
+
+    this(int x)
+    {
+        new(m) Y8(x); // initialise `m`
+    }
```

Comment:
Missing tests for partially constructing array of `Y8`.

Reply:
If you could post some code illustrating what you mean, I'd appreciate it!